apps/demo/src/app/product/pages/product-view/product-view.component.spec.ts

```diff
@@ -115,45 +106,54 @@ describe('ProductViewComponent', () => {
     expect(component).toBeTruthy();
   });
 
-  it('should set productId from id param', () => {
-    expect(component.productId).toEqual(idParam);
-  });
+  describe('ngOnInit', () => {
+    it('should initialize product$', () => {
+      facade.product$ = hot('ab', { a: null, b: mockProduct });
+      component.ngOnInit();
+      let expected = cold('ab', { a: null, b: mockProduct });
+      expect(component.product$).toBeObservable(expected);
+    });
 
-  describe('on [product-container]', () => {
-    
-    it('should set selectedProductId', () => {
-      expect(productContainer.selectedProductId).toEqual(idParam);
+    it('should initialize loading$', () => {
+      facade.loading$ = hot('ab', { a: false, b: true });
+      component.ngOnInit();
+      let expected = cold('ab', { a: false, b: true });
+      expect(component.loading$).toBeObservable(expected);
     });
-  });
+
+    it('should dispatch a  DaffProductLoad with an `id`', () => {
+      spyOn(facade, 'dispatch');
+      component.ngOnInit();
+      expect(facade.dispatch).toHaveBeenCalled();
```

Comment:
should be `expect(facade.dispatch).toHaveBeenCalledWith(new DaffProductLoad(id))`

Reply:
9744558c75652fc3a4b171b5e1bad131e618874d